09_Loading_Elf/02_Loading_And_Running.md

```diff
@@ -14,14 +14,48 @@ In the previous chapter we looked at the details of loading program headers, but
 
 - First a copy of the ELF file to be loaded is needed. The recommended way is to load a file via the VFS, but it could be a bootloader module or even embedded into the kernel.
 - Then once the ELF is loaded, we need verify that its header is correct. Also check the architecture (machine type) matches the current machine, and that the bit-ness is correct (dont try to run a 32-bit program if you dont support it!).
-- Find all the loadable program headers for the ELF, we'll need those in a moment.
+- Find all the loadable program headers for the ELF, we'll need those in a moment (They are the headers with type `PT_LOAD`).
 - Create a new address space for the program to live in. This usually involves creating a new process with a new VMM instance, but the specifics will vary depending on your design. Don't forget to keep the kernel mappings in the higher half!
 - Copy the loadable program headers into this new address space. Take care when writing this code, as the program headers may not be page-aligned:. Don't forget to zero the extra bytes between `memsz` and `filesz`.
 - Once loaded, set the appropriate permission on the memory each program header lives in: the write, execute (or no-execute) and user flags.
 - Now we'll need to create a new thread to act as the main thread for this program, and set its entry point to the `e_entry` field in the ELF header. This field is the start function of the program. You'll also need to create a stack in the memory space of this program for the thread to use, if this wasnt already done as part of our thread creation.
 
 If all of the above are done,  then the program is ready to run! We now should be able to enqueue the main thread in the scheduler and let it run.
 
+### What We Verify?
+
+When veryfyig an ELF file there are few things we need to check in order to decide if an executable is valid, the field to validate are in different headers, some of them are in the `e_ident` header, and they are the following:
```

Comment:
I think veryfyig is meant to be verifying :D
We could also break the sentence to give it a better flow:
> the fields to validate are at different points in the ELF header. Some can be found in the `e_ident` field, like the following: